src/functional/iterator/embedded.py

```diff
@@ -332,9 +399,9 @@ def shift(self, *offsets):
     def max_neighbors(self):
         assert self.incomplete_offsets
         assert isinstance(
-            self.offset_provider[self.incomplete_offsets[0].value], NeighborTableOffsetProvider
+            self.offset_provider[self.incomplete_offsets[0]], NeighborTableOffsetProvider
         )
-        return self.offset_provider[self.incomplete_offsets[0].value].max_neighbors
+        return self.offset_provider[self.incomplete_offsets[0]].max_neighbors
 
     def can_deref(self):
```

Comment:
Missing type hint. Also, would it make more sense to use a `@property` here?

Reply:
As far as I can see `@property` with `Protocol` has the disadvantage that you cannot do `issubclass` checks (though, I currently have no use-case other than testing an iterator implementation).